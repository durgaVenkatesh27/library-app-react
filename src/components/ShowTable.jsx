import React from 'react';
import * as _ from 'lodash';
import { withStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import { MaintainLibraryLocale } from '../locale';
import Button from '@material-ui/core/Button';

const StyledTableCell = withStyles((theme) => ({
  head: {
    backgroundColor: "cornflowerblue",
    color: theme.palette.common.white,
  },
  body: {
    fontSize: 14,
  },
}))(TableCell);

const StyledTableRow = withStyles((theme) => ({
  root: {
    '&:nth-of-type(odd)': {
      backgroundColor: theme.palette.action.hover,
    },
  },
}))(TableRow);

const ShowTable = props => {
  const { tableColumn, tableData, section, functionCall } = props;

  return (
    <div>
      <TableContainer component={Paper} >
        <Table aria-label="customized table" className={section}>
          <TableHead>
            <TableRow>
              {
                tableColumn.map((column, index) => (
                  <StyledTableCell
                    key={index}
                    align={"center"}
                    id={`${column}ColumnHeader`}
                  > {column}</StyledTableCell>
                ))
              }
            </TableRow>
          </TableHead>
          <TableBody>
            {_.isEmpty(tableData) ? " No data " : tableData.map((row, index) => (

              section === 'user' ? (
                <StyledTableRow key={row.id} id="tableBody">
                  <StyledTableCell align="center">{row.id}</StyledTableCell>
                  <StyledTableCell align="center">{row.name}</StyledTableCell>
                  <StyledTableCell align="center">{row.bookList}</StyledTableCell>
                </StyledTableRow>)
                : (section === 'book' ?
                  (<StyledTableRow key={row.id} id="tableBody">
                    <StyledTableCell align="center">{row.id}</StyledTableCell>
                    <StyledTableCell align="center">{row.bookTitle}</StyledTableCell>
                  </StyledTableRow>)
                  : (section === 'borrow' ? (<StyledTableRow key={row.id} id="tableBody">
                    <StyledTableCell align="center">{row.id}</StyledTableCell>
                    <StyledTableCell align="center">{row.bookTitle}</StyledTableCell>
                    <StyledTableCell align="center">
                      <Button
                        variant="contained"
                        size="small"
                        color="primary"
                        style={{ 'text-transform': 'none' }}
                        type="button"
                        className={`borrow${index} primary`}
                        onClick={(e) => functionCall(e, row.id)}
                        id={`borrow${index}`}
                      >
                        {_.get(MaintainLibraryLocale, 'borrowButtonTxt', '')}
                      </Button>
                    </StyledTableCell>
                  </StyledTableRow>)
                    : (<StyledTableRow key={row.id} id="tableBody">
                      <StyledTableCell align="center">{row.id}</StyledTableCell>
                      <StyledTableCell align="center">{row.bookTitle}</StyledTableCell>
                      <StyledTableCell align="center">
                        <Button
                          variant="contained"
                          size="small"
                          color="secondary"
                          style={{ 'text-transform': 'none' }}
                          type="button"
                          onClick={(e) => functionCall(e, row.id)}
                          id={`borrowList${index}`}
                        >
                          {_.get(MaintainLibraryLocale, 'returnButtonTxt', '')}
                        </Button>
                      </StyledTableCell>
                    </StyledTableRow>)

                  ))
            ))}
          </TableBody>
        </Table>
      </TableContainer>
    </div>
  );
}


export default ShowTable;