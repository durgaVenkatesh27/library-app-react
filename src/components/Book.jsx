import React, { useState } from 'react';
import * as _ from 'lodash';
import { BookInfoLocale } from '../locale/index';
import * as api from '../commons/Api';
import ShowTable from './ShowTable';
import '../index.css';
import ErrorAlert from '../commons/ErrorAlert';


const Book = props => {
  const { bookInfo, updateBookData } = props;
  const [bookName, setBookName] = useState('');
  const [alert, setAlert] = useState({ showMsg: false, message: '', severity: 'error' })

  const handleChange = (e) => {
    const { target: { value } } = e;
    alert.showMsg = false;
    setAlert(alert);
    setBookName(value);
  }

  const addBookInfo = (e) => {
    e.preventDefault();

    api.addUser(
      "http://localhost:9090/library/addBooks",
      setBookName,
      { "bookTitle": bookName, "isAvailable": true },
      BookInfoLocale,
      setAlert
    );
    updateBookData();
  }
  return (
    <div>
      {
        _.get(alert, 'showMsg', false)
        && (<ErrorAlert errorMessage={_.get(alert, 'message', '')} errorSeverity={_.get(alert, 'severity', '')} />)
      }
      <form className="maintain-info">
        <label htmlFor="bookName">
          {_.get(BookInfoLocale, 'bookName', '')}
        </label>
        <input
          className="custom-input"
          type="text"
          name="bookName"
          id="bookName"
          value={bookName}
          autoComplete='off'
          onChange={(e) => handleChange(e)}
        />
        <button
          className="custom-button"
          type="submit"
          disabled={bookName.length === 0}
          onClick={(e) => addBookInfo(e)}
          id="addBook"
        >
          {_.get(BookInfoLocale, 'addButtonTxt', '')}
        </button>
      </form>
      <div>
        <ShowTable
          tableColumn={_.get(BookInfoLocale, 'bookTableColumns', '')}
          tableData={bookInfo}
          section='book'
        />
      </div>
    </div>
  );
}

export default Book;