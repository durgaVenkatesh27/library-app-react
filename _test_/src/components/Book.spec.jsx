import React from 'react';
import { mount, shallow } from 'enzyme';
import Book from '../../../src/components/Book';
import ShowTable from '../../../src/components/ShowTable';


describe('Book component Tests', () => {
    let props;
    global.fetch = jest.fn(() =>
        Promise.resolve({
            json: () => Promise.resolve({ rates: { CAD: 1.42 } }),
        }));
    beforeEach(() => {
        props = {
            bookInfo: [{ id: 1, bookTitle: 'Sample', isAvailable: true }],
            updateBookInfo: jest.fn()
        }
    })
    it('should render Book Component', () => {
        const renderedModule = shallow(<Book {...props} />);
        expect(renderedModule).toMatchSnapshot();
    });
    it('should render ShowTable Component', () => {
        const renderedModule = mount(<Book {...props} />);
        expect(renderedModule).toMatchSnapshot();
        expect(renderedModule.find(ShowTable).length).toBe(1);
    });
    it('should add book', () => {
        let book = [];
        const updateBookData = jest.fn(() => {
            book = [{ id: 1, bookTitle: 'Sample', isAvailable: true },
            { id: 1, bookTitle: 'Sample2', isAvailable: true }];
            return true;
        });
        const renderedModule = mount(<Book {...props} updateBookData={updateBookData} />);

        renderedModule.find("#bookName").simulate("change", {
            target: { value: "sample2" }
        });
        renderedModule.find("#addBook").simulate("click");

        expect(renderedModule).toMatchSnapshot();
        expect(renderedModule.find('#bookName').text()).toBe('');
    });
});