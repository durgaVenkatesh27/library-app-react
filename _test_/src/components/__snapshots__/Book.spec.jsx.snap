// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book component Tests should add book 1`] = `ReactWrapper {}`;

exports[`Book component Tests should render Book Component 1`] = `ShallowWrapper {}`;

exports[`Book component Tests should render ShowTable Component 1`] = `ReactWrapper {}`;
